### PR TITLE
added json-c lib and implemented IPC get_version

### PR DESCRIPTION
**Pull request by minus7**
version as far as possible without working on the versioning code

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,12 +22,15 @@ find_package(XKBCommon REQUIRED)
 find_package(WLC REQUIRED)
 find_package(A2X REQUIRED)
 find_package(PCRE REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(JSON REQUIRED json-c)
```
Comment (SirCmpwn) [inline CMakeLists.txt]: Don't do it this way. Write a `FindJsonC.cmake` instead.

**Comment by SirCmpwn** [comment]
As soon as we introduce a JSON library, I want all of our JSON stuff refactored to use it.

File: CMakeLists.txt (hunk 1 above)
Comment (minus7) [inline CMakeLists.txt]: Any particular reason against pkg-config?

Comment (SirCmpwn) [inline CMakeLists.txt]: You can use pkg-config inside the cmake file, I just want consistent dependency resolution between all of our dependencies.

**Pull request merged**